### PR TITLE
Check for the file version before formatting after CDD.

### PR DESCRIPTION
**Pull request by sean-mcmanus**
Fix for https://github.com/microsoft/vscode-cpptools/issues/10203

Check if the document version is modified before applying the formatting edit.

Also, replaced all the nested "ifs" with early returns since the nesting was getting large.

### CODE REVIEW

File: Extension/src/LanguageServer/client.ts [hunk 1]
```diff
@@ -2982,80 +2982,99 @@ export class DefaultClient implements Client {
...
+        // Run formatRange.
+        const formatEdits: vscode.WorkspaceEdit = new vscode.WorkspaceEdit();
+        const formatRange: vscode.Range = new vscode.Range(selectionRange.start, new vscode.Position(selectionRange.start.line + numNewlines, 0));
+        const settings: OtherSettings = new OtherSettings(vscode.workspace.getWorkspaceFolder(modifiedDocument)?.uri);
+        const formatOptions: vscode.FormattingOptions = {
+            insertSpaces: settings.editorInsertSpaces ?? true,
+            tabSize: settings.editorTabSize ?? 4
+        };
+        const versionBeforeFormatting: number | undefined = openFileVersions.get(modifiedDocument.toString());
+        if (versionBeforeFormatting === undefined) {
```
Comment (sean-mcmanus) [inline Extension/src/LanguageServer/client.ts:3061]: The version checking code is here. The rest of the changes is just from early returning.

**Comment by bobbrow** [comment]
Need to resolve with @michelleangela's change which already removed the formatting.

**Comment by sean-mcmanus** [comment]
> Refe

**Comment by sean-mcmanus** [comment]
> Need to resolve with @michelleangela's change which already removed the formatting.

Yeah, I had to submit an empty commit.

**Review by bobbrow — APPROVED** [review]

**Review by michelleangela — APPROVED** [review]

**Pull request merged**